contracts/Depository.sol

```diff
@@ -288,13 +295,19 @@ contract Depository is IErrorsTokenomics {
         // Get the LP token address
         address token = product.token;
 
-        // Check for the product expiry
-        // Note that if the token or productId are invalid, the expiry will be zero by default and revert the function
-        expiry = product.expiry;
-        if (expiry < block.timestamp) {
-            revert ProductExpired(token, productId, product.expiry, block.timestamp);
+        // Check for the product vesting time
+        // Note that if the token or productId are invalid, the vesting will be zero by default and result in revert
+        uint256 vesting = product.vesting;
+        if (vesting == 0) {
+            revert ProductExpired(token, productId, vesting, block.timestamp);
```

Comment:
The product does not expire unless its supply becomes zero, and then it's instantly closed.

Reply:
But here if the product ID is invalid this will be a misleading revert?

Reply:
I'm confused about the whole condition. Why would vesting ever be zero?

Reply:
The vesting is zero when the product is inactive or never existed. The revert is not confusing in a sense that the product can't be used. We are just utilizing the revert that was defined from the previous version as this error is part of the contract anyway. We can redefine another one and call it `ProductInactiveOrNonExistent(productId)`, would that work better?

Reply:
We also have a `ProductClosed(uint256 productId)` error, I think it is good enough.